blueflood-elasticsearch/src/test/java/com/rackspacecloud/blueflood/io/EnumElasticIOTest.java

```diff
@@ -0,0 +1,309 @@
+        enumElasticIO.insertDiscovery(metrics);
+
+        // verify
+        verify(bulkRequestBuilderMock, never()).add(any(IndexRequestBuilder.class));
+    }
+
+    @Test
+    public void testInsertDiscovery() throws IOException {
+        // setup mock for BulkRequestBuilder
+        ListenableActionFuture listenableActionFutureMock = mock(ListenableActionFuture.class);
+        when(bulkRequestBuilderMock.execute()).thenReturn(listenableActionFutureMock);
+
+        // setup mock for ElasticSearch Client
+        IndexRequestBuilder indexRequestBuilderMock = mock(IndexRequestBuilder.class);
+        when(esClientMock.prepareBulk()).thenReturn(bulkRequestBuilderMock);
+        when(esClientMock.prepareIndex(EnumElasticIO.ENUMS_INDEX_NAME_WRITE, EnumElasticIO.ENUMS_DOCUMENT_TYPE))
+                .thenReturn(indexRequestBuilderMock);
+        when(indexRequestBuilderMock.setId(anyString())).thenReturn(indexRequestBuilderMock);
+        when(indexRequestBuilderMock.setSource(any(XContentBuilder.class))).thenReturn(indexRequestBuilderMock);
+        when(indexRequestBuilderMock.setCreate(anyBoolean())).thenReturn(indexRequestBuilderMock);
+        when(indexRequestBuilderMock.setRouting(anyString())).thenReturn(indexRequestBuilderMock);
+        enumElasticIO.setClient(esClientMock);
+
+        // create PreaggregatedMetric metrics list and call insertDiscovery with it
+        List<IMetric> metrics = new ArrayList<IMetric>();
+        long collectionTime = 0;
+        Locator locator1 = Locator.createLocatorFromPathComponents(TENANT_1, METRIC_NAME_A1);
+        Locator locator2 = Locator.createLocatorFromPathComponents(TENANT_2, METRIC_NAME_B1);
+        TimeValue ttl = new TimeValue(1, TimeUnit.DAYS);
+        metrics.add(new PreaggregatedMetric(collectionTime, locator1, ttl, rollupWithEnumValues));
+        metrics.add(new PreaggregatedMetric(collectionTime, locator2, ttl, rollupWithEnumValues));
+        enumElasticIO.insertDiscovery(metrics);
+
+        // verify client and bulk
+        verify(esClientMock, times(1)).prepareBulk();
+        verify(bulkRequestBuilderMock, times(2)).add(any(IndexRequestBuilder.class));
+        verify(esClientMock, times(2)).prepareIndex(EnumElasticIO.ENUMS_INDEX_NAME_WRITE, EnumElasticIO.ENUMS_DOCUMENT_TYPE);
+        verify(bulkRequestBuilderMock, times(1)).execute();
+
+        // verify indexRequestBuilder
+        verify(indexRequestBuilderMock, times(2)).setId(anyString());
+        verify(indexRequestBuilderMock).setId(TENANT_1 + ":" + METRIC_NAME_A1);
+        verify(indexRequestBuilderMock).setId(TENANT_2 + ":" + METRIC_NAME_B1);
+        verify(indexRequestBuilderMock, times(2)).setSource(any(XContentBuilder.class));
+        verify(indexRequestBuilderMock, times(2)).setCreate(true);
+        verify(indexRequestBuilderMock, times(2)).setRouting(anyString());
+        verify(indexRequestBuilderMock).setRouting(TENANT_1);
+        verify(indexRequestBuilderMock).setRouting(TENANT_2);
+    }
+
```

Comment:
We might be.  If the tests started failing because of implementation changes and becoming tedious to maintain, we can just remove these routing and  create call and just assert the locator field.  Or do you want me to remove it now?